src/main/java/org/openlmis/buq/web/buq/BottomUpQuantificationController.java

```diff
@@ -212,6 +230,14 @@ public BottomUpQuantificationDto authorize(@PathVariable("id") UUID bottomUpQuan
       throw new NotFoundException(MessageKeys.ERROR_BOTTOM_UP_QUANTIFICATION_NOT_FOUND);
     }
 
+    ProgramDto program = programReferenceDataService
+            .findOne(bottomUpQuantificationDto.getProgramId());
+    if (program.getSkipAuthorization()) {
```

Comment:
are we sure `program` is not nullable at this point? (in other words - if `findOne` returns `null` if it does not find any program, it would cause a `NullPointerException` here.

Reply:
I think `program` in this case would be null and `program.getSkipAuthorization()` would throw `npe`. good catch, thanks